test-scripts/standalone.sh

```diff
@@ -1,11 +1,69 @@
 #!/bin/bash
 
-# NOTE that this script will hang if the offsets are wrong. In other words,
-# only use this for testing/demo; NOT production.
+# This script sets up and runs a standalone test for a PPM container. The 
+# PPM is started in a Docker container using a specified image and waits for it to become
+# ready. The script takes in three input files: ROAD_FILE, CONFIG, TEST_DATA, and a type
+# argument (BSM or TIM). It checks if the input files exist and copies them to a test data
+# directory. If the OFFSET argument is provided, it is used as the offset in the topic that
+# will be consumed and displayed in the output. If not, the default value of 0 is used.
+
+# The script then produces the test data by executing either do_bsm_test.sh or do_tim_test.sh
+# depending on the type argument, passing in the OFFSET value as an argument. The PPM container
+# is then stopped, and the script ends.
+
+# This script should only be used for testing or demo purposes, as it may hang if the offsets are wrong. It also
+# checks if the required configuration and test data files exist before proceeding with the test.
+
+PPM_CONTAINER_NAME=test_ppm_instance
+PPM_IMAGE_TAG=do-kafka-test-ppm-image
+PPM_IMAGE_NAME=jpo-cvdp_ppm
+
+startPPMContainer() {
+    stopPPMContainer
+    
+    # Start the PPM in a new container.
+    dockerHostIp=$DOCKER_HOST_IP
+
+    # make sure ip can be pinged
+    while true; do
+        if ping -c 1 $dockerHostIp &> /dev/null; then
+            break
+        else
+            echo "Docker host ip $dockerHostIp is not pingable. Exiting."
+            exit 1
+        fi
+    done
+    echo "Starting PPM in new container '$PPM_CONTAINER_NAME'"
+    docker run --name $PPM_CONTAINER_NAME --env DOCKER_HOST_IP=$dockerHostIp --env PPM_LOG_TO_CONSOLE=true --env PPM_LOG_TO_FILE=true -v /tmp/docker-test/data:/ppm_data -d -p '8080:8080' $PPM_IMAGE_NAME:$PPM_IMAGE_TAG /cvdi-stream/docker-test/ppm_standalone.sh
```

Comment:
![image](https://github.com/usdot-jpo-ode/jpo-cvdp/assets/62157949/6a12f6ec-0d33-4267-bdcf-dcfce05b6e01)
Can we use relative path for testing? The container is created but not running.

Reply:
Since the `standalone.sh` script in intended to run inside of a freshly-spun-up docker container, I believe using an absolute path here is the way to go.

The reason that you are experiencing stalling is because the script is expecting the container to have a number of logs before it is deemed 'ready'. Recently, we switched the default logging level to ERROR instead of TRACE, so the container is not spitting out any logs. This should be resolved by https://github.com/CDOT-CV/jpo-cvdp/pull/36. It should be noted that CDOT#36 is not merged into develop yet, so I suggest trying this again once it has been merged.

Reply:
It makes sense  as /cvdi-stream is inside container. Would like to clarify that I am not referring to /cvdi-stream/. I am running the standalone.sh inside my ubuntu, and I do not have this /tmp/docker-test/data directory. I am wandering if we could reuse the data directory in this repository.